flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/api/TableEnvironmentTest.scala

```diff
@@ -75,4 +84,41 @@ class TableEnvironmentTest {
       "  LogicalTableScan(table=[[default_catalog, default_database, MyTable]])\n"
     assertEquals(expected, actual)
   }
+
+  @Test
+  def testExecuteTwiceUsingSameTableEnv(): Unit = {
+    val settings = EnvironmentSettings.newInstance().useBlinkPlanner().inBatchMode().build()
```

Comment:
Does this also works for streaming mode? 
From the issue title and description, it seems that we want to fix it both in streaming mode and batch mode.

Reply:
added `TableEnvironmentITCase` to test batch and stream